### PR TITLE
ifdtool: add an ifdtool (WIP)

### PR DESCRIPTION
**Pull request by rminnich**
Signed-off-by: Ronald G. Minnich <rminnich@google.com>

### CODE REVIEW

File: cmds/ifdtool/dat.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+const (
+	f20MHZ = 0
+	f33MHZ
+	f48MHZ
+	f50_30MHZ = 4
+	f17MHZ    = 6
+)
+
+const (
+	s512KB = 0
```
Comment (rjoleary) [inline cmds/ifdtool/dat.go]: You have to write `iota` here instead of 0, otherwise all the values will equal 0. By default, it doesn't work like c enums.

Comment (rjoleary) [inline cmds/ifdtool/dat.go]: https://play.golang.org/p/ReXMTMmg7_

File: cmds/ifdtool/dat.go [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+	s32MB
+	s64MB
+	sUNUSED = 0xf
+)
+
+type fdbar struct {
+	Map0  uint32
+	Map1  uint32
+	Map2  uint32
+	_     [0xefc - 0x20]uint8
```
Comment (rjoleary) [inline cmds/ifdtool/dat.go]: Nice!

**Comment by rjoleary** [comment]
What is the progress so far?
It doesn't look like it can replace ifdtool or ich_descriptors_tool yet.

File: cmds/ifdtool/dat.go (hunk 1 above)
Comment (GanShun) [inline cmds/ifdtool/dat.go]: yup ryan is right here. surprised me too. You'll want to change for all your constants in this file

File: cmds/ifdtool/ifdtool.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+	if err := binary.Read(c, b, &c.Data); err != nil {
+		return fmt.Errorf("%v", err)
+	}
+	return nil
+
+}
+
+func usage() {
+	log.Fatalf("Usage: yeah right")
```
Comment (GanShun) [inline cmds/ifdtool/ifdtool.go]: wat

**Comment by rminnich** [comment]
Another day.

**Pull request closed without merging**